app/v1/models/request/disclosures/AmendClass2Nics.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.request.disclosures
+
+import play.api.libs.json.{JsPath, Json, OWrites, Reads}
+
+case class AmendClass2Nics(class2VoluntaryContributions: Boolean)
+
+object AmendClass2Nics {
+  implicit val reads: Reads[AmendClass2Nics] = Json.reads[AmendClass2Nics]
+
+  implicit val writes: OWrites[AmendClass2Nics] = (JsPath \ "class2VoluntaryContributions").write[Boolean].contramap(_.class2VoluntaryContributions)
```

Comment:
what is this?

Reply:
This was already in the code. It was used for both AmendDisclosuresRequestBody and RetrieveDisclosuresResponse so  just changed the name to AmendClassNics. To be honest, it can be changed to OFormat[Class2Nics] as we have the DES specs now. This was used case classes with single field.

Reply:
I don't understand why this is a contramap

Reply:
I think because it's case class is one with a single field. contramap is used for case classes with a single field writes and map for case class with single case class reads with Jspath

Reply:
But the field name doesn't change, so surely just a format would be more appropriate?

Reply:
It seems you might not have gotten my first comment. I just changed the name of file for case class Class2Nics to AmendClass2Nics as just the case class was used for both the RetrieveDisclosuresResponse and AmendDisclosuresRequestBody. Didn't look at the content but created a separate one for the Retrieve's Class2Nics which I used OFormat. That's why I said it can be changed as we have the DES specs now. We didn't at the time so the contramap was used for both amend and retrieve at the time.

Reply:
But yh, it can be changed now as we have the DES specs. Didn't know if I needed to change it in this story as we need to for state benefits and income received endpoints as they can be changed to OFormat also. But yh you are right.